src/ai/defaultai_warfare.cc

```diff
@@ -985,30 +985,95 @@ bool DefaultAI::check_militarysites(const Time& gametime) {
+                         -std::abs(management_data.get_military_number_at(99));
+	dismantle_score -= (bf.portspace_nearby == ExtendedBool::kTrue) ? std::abs(management_data.get_military_number_at(91)) : 0;
+
+	dismantle_score += management_data.neuron_pool[28].get_result_safe(
+		                        bf.area_military_capacity, kAbsValue);
+	dismantle_score += management_data.neuron_pool[57].get_result_safe(
+		                        bf.future_area_military_capacity, kAbsValue);
+	dismantle_score -= management_data.neuron_pool[29].get_result_safe(
+		                        bf.military_loneliness / 50, kAbsValue);
+	//dismantle_score -= management_data.neuron_pool[30].get_result_safe( // NOCOM uncomment this
+	//	                        bf.future_military_loneliness, kAbsValue);
+	if (soldier_status_ == SoldiersStatus::kBadShortage) {
+		dismantle_score += 2 * std::abs(management_data.get_military_number_at(77));
+	} else if (soldier_status_ == SoldiersStatus::kShortage) {
+		dismantle_score += std::abs(management_data.get_military_number_at(77));
+	}
+	dismantle_score -= management_data.neuron_pool[45].get_result_safe(
+		                        bf.military_score_ / 50, kAbsValue);
+	dismantle_score += management_data.neuron_pool[46].get_result_safe(
+		                        bf.own_military_presence, kAbsValue);
+	dismantle_score -= management_data.neuron_pool[47].get_result_safe(
+		                        bf.enemy_military_presence, kAbsValue);
+	dismantle_score -= management_data.neuron_pool[48].get_result_safe(
+		                        bf.enemy_military_sites * 3, kAbsValue);
+
+	const bool should_be_dismantled = dismantle_score > 0;
+
+	// const int32_t enemy_military_capacity = std::max<int32_t>(
+	//    {bf.enemy_military_presence,
+	//     bf.enemy_military_sites * (1 + std::abs(management_data.get_military_number_at(77) / 20)),
+	//     (bf.enemy_owned_land_nearby) != 0u ?
+    //       4 + std::abs(management_data.get_military_number_at(99) / 20) :
+    //       0});
+	// if (bf.enemy_owned_land_nearby != 0u) {
+	// 	if (bf.military_score_ < std::abs(management_data.get_military_number_at(91) * 10) &&
+	// 	    bf.future_area_military_capacity - static_cast<int16_t>(total_capacity) -
+	// 	          std::abs(management_data.get_military_number_at(84) / 10) >
+	// 	       (std::abs(management_data.get_military_number_at(24) / 25) + 1) *
+	// 	          enemy_military_capacity) {
+	// 		should_be_dismantled = true;
+	// 	}
+	// } else {
+	// 	const uint16_t size_bonus =
+	// 	   total_capacity * std::abs(management_data.get_military_number_at(89)) / 5;
+	// 	if (bf.military_score_ + size_bonus < management_data.get_military_number_at(88) * 5 &&
+	// 	    bf.future_area_military_capacity > static_cast<int16_t>(total_capacity)) {
+	// 		should_be_dismantled = true;
+	// 	}
+	// }
+
+	if (can_be_dismantled) {
+		printf("dismantle score: %4d, can and should be dismantled: %s\n",
```

Comment:
please use verb_log_debg function for such cases.

Reply:
I plan to remove this altogether, but maybe can stay as debug... I will see